spec/features/posts/index.html.erb_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :feature do
+  before :each do
+    @user = User.create(name: 'Gaby', photo: 'bulma.jpg', bio: 'Systems Engineering Student')
+
+    @post1 = Post.create(author: @user, title: 'Post 1', text: 'Post One')
+    post2 = Post.create(author: @user, title: 'Post 2', text: 'Post Two')
+    post3 = Post.create(author: @user, title: 'Post 3', text: 'Post Three')
+    post4 = Post.create(author: @user, title: 'Post 4', text: 'Post Four')
+    @posts = [post2, post3, post4]
+
+    visit posts_path(@user.id)
+  end
+
+  it "shows user's profile picture" do
+    expect(page).to have_selector("img[src*='#{@user.photo}']")
+  end
+
+  it "shows the user's username" do
+    expect(page).to have_content(@user.name.to_s)
+  end
+
+  it 'shows the number of posts the user has written.' do
+    expect(page).to have_content("Number of posts: #{@user.posts_counter}")
+  end
+
+  it 'shows a post title' do
+    @posts.each do |post|
+      expect(page).to have_content(post.title)
+    end
+  end
+
+  it 'shows part of the post body' do
+    @posts.each do |post|
+      expect(page).to have_content(post.text)
+    end
+  end
+
+  it 'displays the first comments in a post' do
+    comments = @post1.comments
+    comments.each do |comment|
+      expect(page).to have_content(comment.text)
+    end
+  end
+
+  it 'shows comments and likes counts' do
+    expect(page).to have_content("Comments: #{@post1.comment_counter}, Likes: #{@post1.like_counter}")
+  end
+
+  it 'redirects  to user profile after clicking a post' do
+    click_link(@post1.title)
+    expect(page).to have_selector("input[type=submit][value='Like']")
+  end
```

Comment:
- [ ] Good job on testing most of the features requested in the projects requirement. However, you didn't add a tests that asserts you can see a section for pagination if there are more posts than fit on the view as requested in the projects requirements. Kindly add that.